tracer/src/Datadog.Trace/Util/AsyncUtil.cs

```diff
@@ -49,6 +49,38 @@ public static void RunSync(Func<Task> task, CancellationToken cancellationToken)
           .GetAwaiter()
           .GetResult();
 
+    /// <summary>
+    /// Executes an async Task method which has a void return value synchronously
+    /// USAGE: AsyncUtil.RunSync(() => AsyncMethod(), millisecondsTimeout);
+    /// </summary>
+    /// <param name="task">Task method to execute</param>
+    /// <param name="millisecondsTimeout">Timeout in milliseconds</param>
+    public static void RunSync(Func<Task> task, int millisecondsTimeout)
+    {
+        _taskFactory
+           .StartNew(TaskWithTimeoutAsync)
+           .Unwrap()
+           .GetAwaiter()
+           .GetResult();
+
+        Task TaskWithTimeoutAsync()
+        {
+            var runTask = task();
+            return runTask.IsCompleted ? runTask : InternalTaskWithTimeoutAsync(runTask, millisecondsTimeout);
```

Comment:
built-in in .NET 6 (same as above)

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3609/commits/18bfc35e3ce644ce86d4f155a7ffde9667c3a1d8